src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -58,70 +56,47 @@ public Task<ICollection<AttachmentSet>> ProcessAttachmentSetsAsync(ICollection<A
-                var type = assembly.GetType(CodeCoverageAnalysisAssemblyName + "." + CoverageInfoTypeName);
-
-                var methodInfo = type?.GetMethod(MergeMethodName);
-
-                if (methodInfo != null)
-                {
-                    IList<string> filesToDelete = new List<string>(files.Count) { tempFileName };
-
-                    for (int i = 1; i < files.Count; i++)
-                    {
-                        cancellationToken.ThrowIfCancellationRequested();
-                        progressReporter?.Report(100 * i / files.Count);
-
-                        cancellationToken.ThrowIfCancellationRequested();
-                        methodInfo.Invoke(null, new object[] { files[i], outputfileName, tempFileName, true });
-
-                        cancellationToken.ThrowIfCancellationRequested();
-                        File.Copy(tempFileName, outputfileName, true);
-
-                        filesToDelete.Add(files[i]);
-                    }
-
-                    cancellationToken.ThrowIfCancellationRequested();
-                    foreach (string fileName in filesToDelete)
-                    {
-                        File.Delete(fileName);
-                    }
-                }
-
+                // Warning: Don't remove this method call.
+                //
+                // We took a dependency on Coverage.CoreLib.Net. In the unlikely case it cannot be
+                // resolved, this method call will throw an exception that will be caught and
+                // absorbed here.
+                this.MergeCodeCoverageFiles(files, cancellationToken);
                 progressReporter?.Report(100);
-                return outputfileName;
             }
             catch (OperationCanceledException)
             {
-                if (EqtTrace.IsWarningEnabled)
-                {
-                    EqtTrace.Warning("CodeCoverageDataCollectorAttachmentsHandler: operation was cancelled.");
-                }
+                // Occurs due to cancellation, ok to re-throw.
                 throw;
             }
             catch (Exception ex)
             {
-                if (EqtTrace.IsErrorEnabled)
-                {
-                    EqtTrace.Error("CodeCoverageDataCollectorAttachmentsHandler: Failed to load datacollector of type : {0} from location : {1}. Error : {2}", CodeCoverageAnalysisAssemblyName, assemblyPath, ex.ToString());
-                }
+                EqtTrace.Error(
+                    "CodeCoverageDataCollectorAttachmentsHandler: Failed to load datacollector. Error: {0}",
+                    ex.ToString());
             }
 
-            return string.Empty;
+            return files[0];
+        }
+
+        private void MergeCodeCoverageFiles(IList<string> files, CancellationToken cancellationToken)
+        {
+            var coverageUtility = new CoverageFileUtility();
+
+            var coverageData = Task.Run(
```

Comment:
Please make this method async. Do not make any Task.Run inside as this is not required.

Reply:
Done, made it async.